src/request/components/request-detail-panel-webservices.jsx

```diff
@@ -0,0 +1,190 @@
+                    <tr className="table-body-padding table-col-title-group">
+                        <th colSpan="6"></th>
+                        <th className="table-col-featured"></th>
+                    </tr>
+                </tfoot>
+            </table>
+        );
+    }
+});
+
+/**
+ * React class to display service details
+ */
+var ServiceDetailsHeaders = React.createClass({
+    render: function() {
+        var headers = this.props.headers;
+        
+        return (
+            <table className="table">
+                {_.map(headers, function(value, key) {
+                    return (
+                        <tr>
+                            <td className="truncate"><strong>{key}:</strong> {value}</td>
+                        </tr>
+                    );
+                })}
+            </table>
+        );
+    }
+});
+
+/**
+ * React class to display service details
+ */
+var ServiceDetails = React.createClass({
+    render: function() {
+        var requestHeaders = this.props.selectedDataHttpRequestMessage.payload.headers;
+        var responseHeaders = this.props.selectedDataHttpResponseMessage.payload.headers;
+            
+        return (
+            <table className="table table-details">
+                <thead>
+                    <tr>
+                        <th className="table-col-title" width="50%">Request</th>
+                        <th className="table-col-title" width="50%">Response</th>
+                    </tr>
+                </thead>
+                <tbody>
+                    <tr>
+                        <td><ServiceDetailsHeaders headers={requestHeaders} /></td>
+                        <td><ServiceDetailsHeaders headers={responseHeaders} /></td>
+                    </tr>
+                </tbody>
+            </table>
+        );
+    }
+});
+
+module.exports = React.createClass({
+    _requestSelected: function(index) {
+        this.setState({
+            selectedIndex: index
+        });
+    },
+    getInitialState: function() {
+        return {
+            selectedIndex: 0
+        };
+    },
+    render: function () {
+        var request = this.props.request;
+
+        // get messages 
+        var payload = getMessages(request);
+        var dataHttpRequestMessages = payload.dataHttpRequest;
+        var dataHttpResponseMessages = payload.dataHttpResponse;
+        
+        var content = null;
+        if (dataHttpRequestMessages && dataHttpResponseMessages) {
+            var selectedDataHttpRequestMessage = dataHttpRequestMessages[this.state.selectedIndex];
+            var selectedDataHttpResponseMessage = dataHttpResponseMessages[this.state.selectedIndex];
+            
+            // intial processing of messages
+            dataHttpRequestMessages = dataHttpRequestMessages.sort(function(a, b) { return a.ordinal - b.ordinal; });
+            dataHttpResponseMessages = dataHttpResponseMessages.sort(function(a, b) { return a.ordinal - b.ordinal; });
+            
```

Comment:
Making some assumptions here that the number of request/response messages are equal and that the pairs are strictly sequential; if a component makes several requests in parallel, this may not hold up.  But that goes back to our earlier discussions about correlation IDs.  It's probably fine for the purposes of the demo (as I don't think the sample app makes multiple external requests in parallel).